tslearn/matrix_profile.py

```diff
@@ -91,15 +97,27 @@ class MatrixProfile(TransformerMixin,
 
     References
     ----------
-    .. [1] C. M. Yeh, Y. Zhu, L. Ulanova, N.Begum et al. 
-       Matrix Profile I: All Pairs Similarity Joins for Time Series: A 
+    .. [1] C. M. Yeh, Y. Zhu, L. Ulanova, N.Begum et al.
+       Matrix Profile I: All Pairs Similarity Joins for Time Series: A
        Unifying View that Includes Motifs, Discords and Shapelets.
        ICDM 2016.
     """
 
-    def __init__(self, subsequence_length=1, scale=True):
+    def __init__(self, subsequence_length=1, algorithm=None, scale=True):
         self.subsequence_length = subsequence_length
         self.scale = scale
+        self.check_if_algorithm_exists(algorithm)
```

Comment:
I believe this check should rather occur at transform time. Otherwise, if someone inits a MP with a correct algorithm name and then change it before `fit_transform`, we'll have an unexpected behavior.

Reply:
Why not pass the implementation type at transform time? Is it to stay consistent with the scikit-learn method signature?

I'll move the check at transform time and make it private!

Reply:
Yes, this is the reason: we want to stick to sklearn API as much as possible.